app/helpers/shared_items_helper.rb

```diff
@@ -0,0 +1,5 @@
+module SharedItemsHelper
+  def remove_expired_links
+    SharedItem.where('expiration < ?', Time.zone.now).destroy_all
```

Comment:
you should destroy only the item you work with, otherwise it will hangs user's browser